system/jlib/jlog.cpp

```diff
@@ -195,11 +195,10 @@ unsigned getPositionOfField(unsigned logfields, unsigned positionoffield)
 
 // LogMsg
 
-LogMsgJobInfo::~LogMsgJobInfo()
-{
-    if (isDeserialized)
-        free((void *) jobIDStr);
-}
+
+//Save all external job strings in a hash table to avoid a non-trivial destructor for LogMsgJobInfo
```

Comment:
How are destructors for thread-local variables handled? Do they have the same performance penalty issues as non-default constructors?

Reply:
I don't know how they are implemented, there must be a thread hook mechanism that is called on thread destruction.  I don't think there is a big performance penalty.
The reason for the change was more that you couldn't have a constexpr object if the destructor is non-trivial

Reply:
Do we need a constexpr object?

Reply:
Only to avoid a destructor.